### PR TITLE
release-20.1: backport #48229 & #49089

### PR DESCRIPTION
**Pull request by miretskiy**
Backport:
  * 2/2 commits from "storage/cloud: Make google storage reader handle download interrupts." (#48229)
  * 1/1 commits from "storage/cloud: Open data stream early to ensure blob exists" (#49089)

Please see individual PRs for details.

/cc @cockroachdb/release

### CODE REVIEW

**Comment by miretskiy** [comment]
@dt can I have your stamp of approval?

**Comment by dt** [comment]
LGTM if you fix the title

**Comment by miretskiy** [comment]
I don't know why the title was screwed up.  Stamp of approval please?

**Review by dt — APPROVED** [review]

**Pull request merged**
